### PR TITLE
Adds Relaxation, and sponge layer helper functions

### PR DESCRIPTION
**Pull request by glwagner**
Depends on functionality in #744 

Needs:

- [x] tests
- [x] possibly more examples in docstrings

Hopefully `Relaxation` will be helpful.

### CODE REVIEW

File: src/Forcing/relaxation.jl [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+Example
+=======
+
+* Restore a field to a linear z-gradient everywhere on a timescale of one minute.
+
+```julia
+julia> restore_stratification = Relaxation(; rate = 1/60, target = (x, y, z, t) -> z)
+```
+"""
+Relaxation(; rate, mask=onefunction, target=zerofunction) =
```
Comment (ali-ramadhan) [inline src/Forcing/relaxation.jl:31]: Hmmm, why not call this `RelaxingFunction`? It's a constructor for `RelaxingFunction` right? Hmmm, well it constructs a `SimpleForcing` so maybe not. I guess the user only interacts with `Relaxation`.

Comment (glwagner) [reply src/Forcing/relaxation.jl:31]: Right! They are different: `RelaxationFunction` is the function itself, `Relaxation` is the forcing. We could also call it `RelaxingForcing`. I just liked the idea of adding some `Relaxation` to my scripts.

**Comment by glwagner** [comment]
> RelaxingFunction uses keyword arguments. Should GaussianMask and LinearTarget use them too?

I'm up for that, though just to clarify, its `Relaxation` that uses keyword arguments, rather than `RelaxationFunction`. The code looks a little mangled which I think may have caused confusion. I'll fix that.

> RIP #291

Ok this is more general! :-/

**Comment by ali-ramadhan** [comment]
> Ok this is more general! :-/

Haha for sure. Just remembering the fallen PRs.

**Comment by glwagner** [comment]
> Haha for sure. Just remembering the fallen PRs.

#291 #neverforget

File: src/Forcing/relaxation.jl (hunk 1 above)
Comment (glwagner) [reply src/Forcing/relaxation.jl:31]: Indeed! `RelaxationFunction` is the function (actually, a callable object), while `Relaxation` is the forcing. I could also call it `RelaxingForcing` or `SimpleRelaxingForcing`. I just liked the idea of adding some `Relaxation` to my scripts.

**Comment by glwagner** [comment]
On the note of #291, maybe we could also add an abstraction for relaxing at a single discrete level.

@ali-ramadhan I changed `GaussianMask` and `LinearTarget` to use keyword arguments. There is also a test. Let me know what you think.

**Pull request merged**